bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/DynamicContributionContributionItem.java

```diff
@@ -29,12 +34,17 @@ class DynamicContributionContributionItem extends ContributionItem {
 
 	private IMenuListener menuListener = IMenuManager::markDirty;
 
+	private static final String BUNDLE_CLASS_PREFIX = "bundleclass://"; //$NON-NLS-1$
+
+	private IActivityManagerProxy activitySupportProxy;
+
 	/**
 	 * Create the item and associated model;
 	 */
 	public DynamicContributionContributionItem(MDynamicMenuContribution item) {
 		super(item.getElementId());
 		model = item;
+		initializeAcivitySupportProxy();
```

Comment:
For E4 ist should be possible to inject services so no need to fetch the service manually,

Reply:
Same here: please propose how exactly this should be done.

Reply:
In `MenuManagerRenderer` (that creates this class) you can do

```
@Inject
IActivityManagerProxy activityManager;
```

then pass the injected instance via the constructor.

Reply:
Ok, please check latest patch.